backend/src/controllers/auth/register.services.ts

```diff
@@ -0,0 +1,13 @@
+import User from '../../models/user.model'
+import { BadRequestError } from '../../errors'
+
+export const createUser = async (userData: any) => {
+    // const userAlreadyExists = await User.findOne(userData.email)
+
+    // if (userAlreadyExists) {
+    //     throw new BadRequestError('Email already in use')
+    // }
+
+    const user = await User.create(userData)
+    return user   
+}
```

Comment:
suggestion, make this `User.service.ts`.
move all business logic here.

```suggestion
import User from '../../models/user.model'
import { BadRequestError } from '../../errors'
import { IRegisterRq } from './auth.interface'

export async function createUser(userData: IRegisterRq){
    return await User.create(userData);
}

export function validate (rq: IRegisterRq){
    if (!rq.username || !rq.email || !rq.password) {
        throw new BadRequestError('Please provide all values');
    }
}


```